pandas/core/groupby/grouper.py

```diff
@@ -374,7 +374,10 @@ def _set_grouper(self, obj: FrameOrSeries, sort: bool = False):
         # possibly sort
         if (self.sort or sort) and not ax.is_monotonic:
             # use stable sort to support first, last, nth
-            indexer = self.indexer = ax.argsort(kind="mergesort")
+            # TODO: why does putting na_position="first" fix datetimelike cases?
```

Comment:
I am not really familiar with the history of NaTs with respect to resample. I would have assumed NaT would/should have been dropped by this point since we just exclude them from the result https://github.com/pandas-dev/pandas/pull/13164. 

I'm okay with this solution as a stop gap solution.